### PR TITLE
update top level Signed.Signed to be a *json.RawMessage

### PR DESCRIPTION
**Pull request by endophage**
Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

File: server/timestamp/timestamp.go [hunk 1]
```diff
@@ -130,13 +130,14 @@ func CreateTimestamp(gun string, prev *data.SignedTimestamp, snapshot []byte, st
 	if prev != nil {
 		ts.Signed.Version = prev.Signed.Version + 1
 	}
-	sgndTs, err := json.MarshalCanonical(ts.Signed)
+	var sgndTs json.RawMessage
+	sgndTs, err = json.MarshalCanonical(ts.Signed)
```
Comment (HuKeping) [inline server/timestamp/timestamp.go]: Why replace the `sgndTs, err := json.MarshalCanonical(ts.Signed)` here?

Comment (endophage) [inline server/timestamp/timestamp.go]: If I don't do it this way, `sgndTs` has type `[]byte` and I have to do a really ugly type cast from `[]byte` to `*json.RawMessage` on the line that changed just below. This is only one line longer and at least in my opinion much cleaner.

Comment (HuKeping) [inline server/timestamp/timestamp.go]: Prefer this way too :+1:

**Comment by riyazdf** [comment]
changes LGTM, also tested against @cyli's integration test and my own local repos (both new and old) with no issues.

**Comment by cyli** [comment]
LGTM!

**Pull request merged**